packages/schema/src/Schema.ts

```diff
@@ -2335,7 +2335,11 @@ const intersectUnionMembers = (
   AST.Union.make(
     xs.flatMap((x) => {
       return ys.map((y) => {
-        if (AST.isTypeLiteral(x)) {
+        if (AST.isUnion(x)) {
+          return intersectUnionMembers(x.types, [y], path)
```

Comment:
Notice that this will create sub-unions, we could un-nest them, see https://github.com/Effect-TS/effect/pull/2398